wrangler-service/src/main/java/co/cask/wrangler/service/directive/DirectivesService.java

```diff
@@ -1002,37 +934,26 @@ public void artifacts(HttpServiceRequest request, HttpServiceResponder responder
    * This HTTP endpoint is used to retrieve plugins that are
    * of type <code>Directive.Type</code> (directive). Artifact will be reported
    * if it atleast has one plugin that is of type "directive".
+   *
+   * TODO: (CDAP-14694) Doesn't look like this should exist. Why wasn't the CDAP endpoint used?
    */
   @GET
   @Path("directives")
   public void directives(HttpServiceRequest request, HttpServiceResponder responder) {
-    JsonObject response = new JsonObject();
-    JsonArray values = new JsonArray();
+    List<DirectiveDescriptor> values = new ArrayList<>();
     try {
       List<ArtifactInfo> artifacts = getContext().listArtifacts();
       for (ArtifactInfo artifact : artifacts) {
         Set<PluginClass> plugins = artifact.getClasses().getPlugins();
-        JsonObject object = new JsonObject();
-        object.addProperty("name", artifact.getName());
-        object.addProperty("version", artifact.getVersion());
-        object.addProperty("scope", artifact.getScope().name());
+        DirectiveArtifact directiveArtifact = new DirectiveArtifact(artifact);
         for (PluginClass plugin : plugins) {
           if (Directive.TYPE.equalsIgnoreCase(plugin.getType())) {
-            JsonObject directive = new JsonObject();
-            directive.addProperty("name", plugin.getName());
-            directive.addProperty("description", plugin.getDescription());
-            directive.addProperty("type", plugin.getType());
-            directive.addProperty("class", plugin.getClassName());
-            directive.add("artifact", object);
```

Comment:
this does not include `properties` right?

Reply:
good catch, added another constructor that doesn't set properties.